2016/presentations/inteligentny_dom_z_(micro)python/text.md

```diff
@@ -0,0 +1,89 @@
+* Raspberry Pi - centrum zarządzania budynkiem,
+* Moduły ESP-12F - bezprzewodowa komunikacja z czujnikami i urządzeniami wykonawczymi,
+* Konwerter USB-UART - komunikacja przez port szeregowy,
+* Moduł przekaźnika RM0 - umożliwia włączanie/wyłączanie urządzeń 230V,
+* Moduł DS18b20 - termometr 1-wire,
+* Zasilacz 5V oraz 3.3V.
+
+Już tak krótka lista pozwoli na sterowanie oświetleniem czy sprawowanie kontroli nad temperaturą w domu. Na początku do testowania funkcjonalności nowych modułów elektronicznych można wykorzystać jedną z platform prototypowych np. Arduino.
+
+## MicroPython i moduł ESP8266
+MicroPython to implementacja języka Python 3, która zawiera podstawową część biblioteki standardowej Python i jest zoptymalizowana pod kątem działania na mikrokontrolerach [micropython]. Istnieje kilka platform sprzętowych pozwalających na uruchomienie środowiska MicroPython takich jak pyBoard czy pyMCU. Układem na którym oprzemy naszą instalację inteligentnego domu będzie ESP8266 [esp8266], a konkretnie moduł o nazwie ESP-12F.
+Co oferuje nam urządzenie wielkości znaczka pocztowego (24 x 16 mm) za niecałe 2$? Oto podstawowe cechy modułu ESP-12F:
+- 32 bit RISC CPU - Tensilica Xtensa LX106 @80 MHz,
+- 11 wyprowadzeń GPIO - wejścia/wyjścia cyfrowe,
+- 1 wejście ADC - 10 bitowy przetwornik analogowo-cyfrowy,
+- interfejsy - SPI, I2C, 2 x UART,
+- WiFi 802.11 b/g/n (2.4 GHz),
+- 4 MB flash.
+
+Komunikacja z modułem ESP-12F odbywa się za pomocą portu szeregowego. Należy zwrócić uwagę że komponent zasilany jest napięciem 3.3V i pracuje na logice o tym samym potencjale.
+
+Oryginalne oprogramowanie modułu obsługuje komunikację za pomocą komend AT. W celu uruchomienia środowiska MicroPython należy wgrać nowy firmware. Najlepszym sposobem uzyskania oprogramowania jest zbudowanie najnowszej wersji ze źródeł. Szczegółowy opis tego procesu znajduje się w repozytorium z kodem źródłowym [micropythonsource], można również wykorzystać znalezione w sieci gotowe pliki binarne firmware. Kolejnym krokiem jest wyczyszczenie pamięci układu i wgranie nowego firmware. Do wykonania tych czynności wykorzystamy narzędzie esptool, które służy do komunikacji z bootloaderem w układzie ESP8266 [esptool]. Przed wpisaniem podanej poniżej komendy należy pamiętać o uruchomieniu modułu ESP w trybie programowania (flash mode), wykonuje się to przez podłączenie pinu GPIO0 do masy.
+
+    $ esptool.py --port /dev/tty.xxx --baud 460800 write_flash --flash_size=8m 0 firmware-combined.bin
+
+W przypadku powodzenia naszym oczom powinien ukazać się następujący rezultat:
+
+    Connecting...
+    Erasing flash...
+    Writing at 0x0007cc00... (100 %)
+
+    Leaving...
+
+Moduł jest gotowy do uruchomienia środowiska MicroPython. Po podłączeniu pinu GPIO0 do stanu wysokiego (3.3V) i zrestartowaniu modułu bootloader załaduje nowy wsad. Możemy już uzyskać dostęp do REPL (Python prompt) poprzez port szeregowy.
```

Comment:
Nie trzeba podciągać pinu (choć jest to zalecane gdy pinu używamy do innych rzeczy, ale wtedy przez opornik, a nie bezpośrednio), wystarczy go odłączyć od masy, bo moduł ma wbudowane oporniki podciągające.